src/agg/process_markers_symbolizer.cpp

```diff
@@ -128,8 +128,8 @@ struct agg_markers_renderer_context : markers_renderer_context
 
                     // Build local transformation matrix by resetting translation coordinates
                     agg::trans_affine marker_tr_copy(marker_tr);
-                    marker_tr_copy.tx = sample_x / sampling_rate - x0;
-                    marker_tr_copy.ty = sample_y / sampling_rate - y0;
+                    marker_tr_copy.tx = dx - x0;
+                    marker_tr_copy.ty = dy - y0;
```

Comment:
Seems this change is the one introducing inconsistencies.

dx comes from:
```double dy = marker_tr.ty - std::floor(marker_tr.ty);```

I think the issue is that, for negative values we want the value closer to 0, eg for `-10.9` we want `-10`, not `-11`. That is achieved using trunc instead of floor. I'll give it a try

Reply:
Tested and it's not the cause :disappointed: I'll probably just revert this change and check everything looks right